backend/packages/Upgrade/src/database/1725913309740-migrations.ts

```diff
@@ -0,0 +1,26 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class Migrations1725913309740 implements MigrationInterface {
+    name = 'Migrations1725913309740'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE "experiment" ADD "moocletDetails" json`);
+        await queryRunner.query(`ALTER TYPE "public"."experiment_assignmentalgorithm_enum" RENAME TO "experiment_assignmentalgorithm_enum_old"`);
+        await queryRunner.query(`CREATE TYPE "public"."experiment_assignmentalgorithm_enum" AS ENUM('random', 'stratified random sampling', 'uniform_random', 'ts_configurable')`);
```

Comment:
Should the algorithm types be constrained by an enum in the UpGrade db? It seems like that would limit the flexibility of being able to add/define different algorithms in the mooclet server.

Reply:
it's definitely one of the things I'm looking for feedback on. It sure is convenient for several reasons to be able to attach an algorithm type enum to the experiment, it feels like that's where it has to go. Each algorithm we want to add support for later will require at least some minimal code updates because each one has a special set of experiment design options called "policy-parameters" that we'll want to validate, and potentially some bespoke usage and testing, so it will need to be a deliberate code change task anyway to include each supported algorithm.

Reply:
I did notice that you could add algorithms ("policies") arbitrarily in the django mooclet interface, but there's no way to add the corresponding function without changing the API code. That said, we can still have enums in the application without enforcing them in the database schema. An argument for that would be: if we wanted to support adding and removing the ones that are already defined in the mooclet, it would be easier if it didn't always have to involve a database migration - especially in the case of removing algorithms.